crates/fluvio-version-manager/src/command/switch.rs

```diff
@@ -0,0 +1,221 @@
+//! Version Switching Command
+//!
+//! The `switch` command is responsible of changing the active Fluvio Version
+
+use std::fs::{create_dir_all, read_dir, copy};
+use std::path::{PathBuf, Path};
+
+use anyhow::Result;
+use clap::Parser;
+use colored::Colorize;
+
+use fluvio_hub_util::fvm::Channel;
+
+use crate::GlobalOptions;
+use crate::common::manifest::{PACKAGE_SET_MANIFEST_FILENAME, VersionManifest};
+use crate::common::settings::Settings;
+use crate::common::workdir::{fvm_versions_path, fluvio_binaries_path};
+use crate::common::notify::Notify;
+
+/// Represents the contents of a version directory (`~/.fvm/versions/<version>`)
+/// where binaries and the manifest are stored.
+///
+/// The `VersionDirectory` represents the contents of a version directory
+/// in 2 parts:
+///
+/// - `contents`: Every file in the directory except for the manifest
+/// - `manifest`: The manifest file
+struct VersionDirectory {
+    contents: Vec<PathBuf>,
+    manifest: PathBuf,
+}
+
+impl VersionDirectory {
+    /// Opens a directory and returns a `VersionDirectory` build from scanning it
+    fn open(path: PathBuf) -> Result<Self> {
+        let mut dir = Self {
+            contents: vec![],
+            manifest: PathBuf::new(),
+        };
+
+        for entry in read_dir(path)? {
+            let entry = entry?;
+
+            if entry.metadata()?.is_dir() {
+                continue;
+            }
+
+            let entry_path = entry.path();
+
+            if entry_path.file_name().ok_or(anyhow::anyhow!(
+                "Failed to get filename from path: {}",
+                entry_path.display()
+            ))? == PACKAGE_SET_MANIFEST_FILENAME
+            {
+                dir.manifest = entry_path;
+            } else {
+                dir.contents.push(entry_path);
+            }
+        }
+
+        if dir.manifest == PathBuf::new() {
+            return Err(anyhow::anyhow!(
+                "Failed to find manifest file in version directory"
+            ));
+        }
+
+        Ok(dir)
```

Comment:
I was considering using `Option<PathBuf>` in the `VersionManifest` struct but it wasnt the actual purpose because I expect `manifest.json` to be present always.